hw/top_earlgrey/ip/ast/doc/_index.md

```diff
@@ -944,91 +1017,122 @@ used for test and calibration purposes and are not required for runtime
 operation. See the [<u>Interface Signals
 Table</u>](#interface-signals-table) for more details.
 
+## AST registers initialization during boot.

```

Comment:
@alphan 
@tjaychen 

Can you have a look to check whether this description aligns with your understanding of how AST should be initialized during boot?

Reply:
Hmm, I thought we were on the same page about this: 
* ROM only copies the configuration in OTP to AST registers in order, see [here](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom/rom_start.S;l=183).
* ROM does not perform any additional checks except for checking ast_init [here](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom/rom.c;l=159).
* Checking alert registers in ROM is TBD.

Reply:
@alphan just to clarify, are you referring to the alert register check only, or also the digest check?

looking at the [assembly](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom/rom_start.S;drc=6b0d4ddebf7d1468d998ea8836679a834286e06c;l=190) that copies the REGA registers from OTP to AST, I think we do not have a digest check at the moment.

Reply:
yes, just the alert register check only, i.e. checking if an active alert.

Reply:
i think we're still in sync. We went back and forth on the digest check a lot, and in the end we have a pretty simple compare since most of the smarts is inside ast. 

The alert checking I think really is just an extension of the rom alert setting based on OTP, so that should be fine as well.

Reply:
The TEST_UNLOCK / RMA state we also went back and forth on a lot.
In TEST state we already don't react to the alert stuff, and the RMA behavior can also be codified in OTP.

So I think we are fairly aligned to the recommended behavior.

Reply:
I think I misunderstood this part:

> It is recommended for the ROM code to condition the copy by a digest verification of the register values. If such a digest is too complicated, a simple tag can be used to condition the copy instead.

ATM we don't perform any checks after writing to AST registers except for checking the done bit.

Reply:
Ah, I totally misunderstood what "tag" means. SGTM.